crates/sui/src/sui_commands.rs

```diff
@@ -498,33 +498,43 @@ impl SuiCommand {
                         // (e.g., testnet, mainnet) from the Move.lock under automated address management.
                         // In addition, tree shaking also requires a network as it needs to fetch
                         // on-chain linkage table of package dependencies.
-                        let config =
-                            client_config.unwrap_or(sui_config_dir()?.join(SUI_CLIENT_CONFIG));
-                        prompt_if_no_config(&config, false).await?;
-                        let context = WalletContext::new(&config, None, None)?;
-
-                        let Ok(client) = context.get_client().await else {
-                            bail!("`sui move build --dump-bytecode-as-base64` requires a connection to the network. Current active network is {} but failed to connect to it.", context.config.active_env.as_ref().unwrap());
-                        };
-                        let read_api = client.read_api();
-
-                        if let Err(e) = client.check_api_version() {
-                            eprintln!("{}", format!("[warning] {e}").yellow().bold());
-                        }
-
-                        let chain_id = if build.ignore_chain {
+                        let (chain_id, client) = if build.ignore_chain {
                             // for tests it's useful to ignore the chain id!
-                            None
+                            (None, None)
```

Comment:
Might be worth printing a warning in this case that (a) we're using the latest known versions of the implicit dependencies, and (b) the resulting binary will have more dependencies because of the `--ignore-chain`?